File: ptra-server/src/utils/models/product.model.ts
export default class ProductModel {
  id: number;
  name: string;
  categoryId: number;
  price: number;

  /**
   * Network URL from Google 😃
   */
  imageURL: string;

  constructor(
    id: number,
    name: string,
    categoryId: number,
    price: number,
    imageURL: string,
  ) {
    this.id = id;
    this.name = name;
    this.categoryId = categoryId;
    this.price = price;
    this.imageURL = imageURL;
  }
}

File: ptra-mobile/lib/features/home/data/models/product_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../../domain/domain.dart';

part 'product_model.freezed.dart';
part 'product_model.g.dart';

@freezed
class ProductModel with _$ProductModel implements IProductEntity {
  factory ProductModel({
    required final int id,
    required final String name,
    required final int categoryId,
    required final double price,
    required final String imageURL,
  }) = _ProductModel;

  factory ProductModel.fromJson(Map<String, dynamic> json) => _$ProductModelFromJson(json);
}
